src/Symfony/Component/Cache/Adapter/TagAwareAdapter.php

```diff
@@ -48,7 +48,6 @@ function ($key, $value, CacheItem $protoItem) {
                 $item->value = $value;
                 $item->defaultLifetime = $protoItem->defaultLifetime;
                 $item->expiry = $protoItem->expiry;
-                $item->innerItem = $protoItem->innerItem;
```

Comment:
here the tags cache item was re-using the innerItem of the original item which causes weird behavior inside the proxy adapter.

Should we set a different innerItem here somehow?

Not sure if the fix should be inside `TagAwareAdapter` or inside `ProxyAdapter`